tuner/tuner/libtuner.py

```diff
@@ -236,20 +237,52 @@ class ParsedDisptachBenchmarkResult:
 class IREEBenchmarkResult:
     # Default format follows output of iree-benchmark-module
     candidate_id: int
-    result_str: str
 
-    def get_mean_time(self) -> Optional[float]:
-        if not self.result_str:
-            return None
-        pattern = r"process_time/real_time_mean\s+([\d.]+)\s\w{2}"
-        match = re.search(pattern, self.result_str)
-        if not match:
+    # A list of dictionaries, each representing a benchmark result
+    # Each dictionary contains fields like: real_time: float, cpu_time: float, time_unit: str, repetitions: int, etc.
+    result_json: list[dict[str, Any]]
+
+    def get_mean_time_us(self) -> Optional[float]:
+        """Compute the mean time (in microseconds) for all of the benchmarks"""
+        if not self.result_json:
             return None
-        try:
-            return float(match.group(1))
-        except ValueError:
+
+        total_time = 0.0
+
+        for benchmark in self.result_json:
+            real_time = benchmark.get("real_time")
+            time_unit = benchmark.get("time_unit")
```

Comment:
We should extract the time only from one result that contains: `"aggregate_name" : "mean"` instead of averaging all entries.